refactor: 비교하기 페이지 요약 탭 리팩토링

JaneChun opened the pull request:
<!-- 이 PR을 요약한 내용으로 위 제목 폼을 채워 주세요. -->
![PR-banner](https://github.com/user-attachments/assets/e9a3d5a0-d8ab-4486-ac39-38b59c211172)

## 작업 개요
**비교하기 페이지 리팩토링** (`apps/knockdog/app/compare-complete-page.tsx`)
- Figma 디자인에 맞춰 스타일 수정
- FSD 아키텍처에 맞게 관심사 분리
- API 통합
 
## 작업 내용
**아키텍처**
  - `entities/compare/*`
    - `api`: 비교 데이터 요청 API
    - `config`: Tanstack Query 키 생성 함수
    - `model`: 비교 도메인 관련 타입 및 상수
        - `compare.ts`: 유치원 비교 도메인 모델
        - `compare-result.ts`: 비교 결과 타입
        - `constants/compare.ts`: 도메인 상수
    - `lib`: 비교 도메인 관련 유틸 함수
    - `ui`: Label, Badge, CircleAvatar 등 재사용 가능한 최소 단위의 공통 UI 컴포넌트

  - `features/compare/*`
    - `api`: Tanstack Query를 사용한 useComparisonsQuery 훅
    - `lib`: 두 유치원의 이용요금, 거리, 공휴일, 픽드랍 등 데이터를 비교하는 비즈니스 로직
    - `ui`: PricingSection, DistanceSection 등 더 기능 단위의 UI 컴포넌트

---

**컴포넌트 상세 설명**
1. Entity 레벨 컴포넌트
   - 텍스트: `Summary`, `Description`, `Detail`, `HighlightedText`
   - 아바타: `CircleAvatar`, `StackedCircleAvatar`
   - 기타: `Label`, `Badge`

2. Feature 레벨 컴포넌트
   - `PricingSection`: 이용요금 비교 섹션
     - `PricingSummary`: "A 유치원이 1시간당 50,000원 더 저렴해요"와 같은 한 줄 요약
     - `PriceDetailedItem`: 정기권, 횟수권별 이용요금 항목을 상세 비교 (`ComparisonDetailedItem` 사용)

   - `DistanceSection`: 거리 비교 섹션
     - `DistanceSummary`: "집에서 A 유치원이 도보로 가장 가까워요"와 같은 한 줄 요약
     - `DistanceDetailedItem`: 도보, 자동차, 대중교통별 소요 시간을 상세 비교(`ComparisonDetailedItem`
       사용)

    - `PickdropSection`: 픽드랍 서비스 비교 섹션 (`ComparisonSimpleItem` 사용)
    - `HolidaySection`: 공휴일 운영 여부 비교 섹션 (`ComparisonSimpleItem` 사용)

    - `OperatingDaysSection` 영업일 비교 섹션 (`ComparisonDaysItem` 사용)

JaneChun committed:
fix: Product 타입의 속성명을 백엔드 응답과 일치하도록 변경

JaneChun committed:
refactor: 아바타 컴포넌트를 공통 UI 컴포넌트를 사용하도록 수정

JaneChun committed:
refactor: Label 컴포넌트 스타일 수정 및 툴팁 UI 추가

JaneChun committed:
refactor: Pill 컴포넌트를 Badge로 이름 변경 및 아이콘/캡션 props 추가

JaneChun committed:
refactor: SelectedCell 컴포넌트 props 및 스타일 수정

JaneChun committed:
Merge branch 'develop' of github.com:PetCampus-Inc/daeng_v2_front into refactor/compare-complete-summary

JaneChun committed:
refactor: 요약 탭 컴포넌트 시스템 재구성

- 텍스트 컴포넌트 : Heading, Description, Detail (+ Highlight 변형)
- 아바타 컴포넌트 : OverlappingAvatars
- 레이아웃  컴포넌트 : ComparisonPanel, CompariosnSection
- 비교 아이템 컴포넌트 :
   - ComparisionDetailedItem: 이용 요금, 거리
   - ComparisonSimpleItem: 픽드랍 서비스, 공휴일
   - ComparisonDaysItem: 영업일
- 기존 SummaryDistanceRow, SummaryDays 제거

JaneChun committed:
feat: 유치원 카테고리 타입 추가 및 serializeCategories 유틸리티 함수 추가

JaneChun committed:
refactor: SelectedCell API 응답 데이터 통합

JaneChun committed:
refactor: 텍스트 표시 UI 컴포넌트 추출

JaneChun committed:
refactor: 아바타 표시 UI 컴포넌트 추출

JaneChun committed:
refactor: 비교 기능 관련 타입을 entities/compare/model/types.ts로 이동

JaneChun committed:
refactor: 비교 페이지 mock 데이터를 entities/compare/model/mock.ts 로 이동

JaneChun committed:
refactor: 비교 페이지 유틸 함수를 entities/compare/lib/utils.ts로 이동

JaneChun committed:
refactor: 레이블, 뱃지 UI 컴포넌트 추출

JaneChun committed:
refactor: 이용요금 및 거리 비교를 위한 타입, 유틸 함수 추가

JaneChun committed:
feat: 이용요금 및 거리 비교 로직, UI 구현

JaneChun committed:
fix: 이용요금 비교 상세 항목에서 금액 표시 오류 수정

JaneChun committed:
feat: 픽드랍 비교 로직 구현 및 UI 컴포넌트 분리

JaneChun committed:
fix: 잘못된 타입명 수정 및 console.log 문 제거

JaneChun committed:
feat: 공휴일 비교 로직, UI 구현

JaneChun committed:
feat: 영업일 비교 로직 구현, UI 컴포넌트 분리

JaneChun committed:
refactor: 비교 페이지 데이터 페칭 로직 리팩토링 (useComparisonsQuery)

JaneChun committed:
chore: 비교 페이지 mock 데이터 삭제

JaneChun committed:
refactor: 상세 비교 아이템 타입 통합

- `PricingDetailedComparison`과 `DistanceDetailedComparison`에서 사용하던 상세 비교 아이템 타입을 `DetailedComparisonItem`으로 통합하여 중복 제거
- 거리 비교 로직이 분(number) 값을 직접 사용하도록 수정하고, 화면에 표시할 때만 시간 문자열로 변환하도록 변경

JaneChun committed:
fix(kindergarten): 잘못된 export문(serializeCategories) 제거

JaneChun committed:
fix: 이용요금 비교 상세 항목에서 금액이 동일할 경우 유치원 정보가 바뀌어 표시되는 오류 수정

JaneChun committed:
fix: 공휴일 비교 로직 오류 수정

JaneChun committed:
refactor: 영업일 관련 상수/타입 리팩토링

JaneChun committed:
fix: 유효하지 않은 Tailwind 클래스 수정

JaneChun committed:
refactor: 최단 거리 계산 로직 안정성 개선

JaneChun committed:
feat: 한글 조사 처리 유틸리티 추가 및 적용

JaneChun committed:
Merge branch 'develop' of github.com:PetCampus-Inc/daeng_v2_front into refactor/compare-complete-summary

yeomgahui commented on apps/knockdog/src/entities/compare/index.ts:
```diff
@@ -0,0 +1,17 @@
+// lib
+export * from './lib/serialize';
```
저희가 컨벤션이 있는데, (저희도 개발 중에 정해서 지금 모든 파일에 적용되어있진 않은데..)
https://www.notion.so/FSD-259ef0acc18a80cba487cae6b1c82d01

여기 참고 부탁드려요 

export의 경우 * 보다는 export { serializeCategories } from ... 
이렇게 명확히 명시해 주기로 했거든요..!

JaneChun replied:
수정 완료했습니다!

yeomgahui commented on apps/knockdog/src/features/compare/ui/DistanceSection.tsx:
```diff
@@ -0,0 +1,40 @@
+import { DistanceDetailedItem } from './DistanceDetailedItem';
+import { Label, DistanceSummary, Badge } from '@entities/compare';
+import { TransportationType, TRANSPORTATION_TYPE, DistanceComparisonsByRef } from '@entities/compare/model/types';
```
features에서 entities 슬라이스 접근시 
@entities/compare 로만 접근 하도록 해야할것 같아요!

JaneChun committed:
fix: 거리 한줄 요약의 잘못된 조사 표시 오류 수정

JaneChun committed:
fix: Header 컴포넌트의 불필요한 withSpacing prop 제거

JaneChun committed:
fix(kindergarten): 잘못된 export문(CTag) 제거

JaneChun committed:
refactor: 이용요금 비교 로직을 PricingSection 내부로 이동

JaneChun committed:
refactor: 거리 비교 로직을 DistanceSection 내부로 이동 및 데이터 구조 개선

JaneChun committed:
refactor: 픽드랍 섹션을 컴포넌트로 분리

JaneChun committed:
refactor: 공휴일 섹션을 컴포넌트로 분리

JaneChun committed:
refactor: 영업일 섹션을 컴포넌트로 분리

JaneChun committed:
refactor: 타입/상수 분리 및 단일 진입점 패턴으로 개선

- 도메인 모델(compare.ts)과 비교 결과(compare-result.ts) 타입 분리
- 상수를 constants/compare.ts로 분리
- features/compare 에서 단일 진입점 사용하도록 import문 수정

JaneChun committed:
fix: onPointerDownOutside 이벤트 핸들러의 매개변수에 Event 타입 추가하여 빌드 타입 오류 해결

WantAirpod commented:
@JaneChun 첫 반영 축하드려요~
